processor/transformprocessor/internal/common/logs.go

```diff
@@ -38,6 +39,7 @@ func (l logStatements) Capabilities() consumer.Capabilities {
 }
 
 func (l logStatements) ConsumeLogs(ctx context.Context, ld plog.Logs) error {
+	var errors error
```

Comment:
Thanks for the explanation. I'm good with just adopting this approach and possibly revisiting configuration later, but I do think we should document this so users are aware of it.

> For your concern, I think users could get into the same situation today since the transform processor doesn'[t undo any modifications when there is an error. If some statements worked and then there is a failure the data is still in an uncompleted state and then passed on to the next processor.

I tested this, and it looks like right now the entire telemetry item is dropped if the transform processor has an error. I dug into it and the processor helper will only invoke the next consumer if the processing function [doesn't return an error](https://github.com/open-telemetry/opentelemetry-collector/blob/d0fe772781ebf6ee038fed296a1bf2cb345d192b/processor/processorhelper/traces.go#L64). Are you seeing different behavior?

Reply:
Did some testing and confirmed this PR's implementation didn't help anything since data is not passed on.  Instead updated the PR to better log errors so that users who encounter errors can take action.